File: F/3162-find-the-number-of-good-pairs-i/main.go
func numberOfPairs(nums1 []int, nums2 []int, k int) int {
	for i := range nums2 {
		nums2[i] = nums2[i] * k
	}

	ans := 0
	for _, s1v := range nums1 {
		for ,s2v := range nums2 {
			if s1v %s2v == 0 {
				ans++
			}
		}
	}
	return ans
}

File: F/3162-find-the-number-of-good-pairs-i/main.rs
struct Solution {}

use std::collections::HashMap;
impl Solution {
  pub fn number_of_pairs(nums1: Vec<i32>, nums2: Vec<i32>, k: i32) -> i32 {
    let mut m: HashMap<i32, i32> = HashMap::new();
    nums2.iter().for_each(|&v| {
      let mut b = v * k;
      while b <= 50 {
        m.entry(b).and_modify(|x| *x += 1).or_insert(1);
        b += v * k;
      }
    });

    nums1
      .iter()
      .map(|x| {
        if m.contains_key(x) {
          *m.get(x).unwrap()
        } else {
          0
        }
      })
      .sum()
  }
}
